openedx/features/subscriptions/models.py

```diff
@@ -106,3 +108,67 @@ def get_valid_subscriptions(self, site, username=None):
             logger.warning('Valid subscription count exceeds 1.')
 
         return user_subscriptions.filter(subscription_id__in=valid_user_subscriptions_ids).order_by('-created')
+
+
+class UserSubscriptionHistory(TimeStampedModel):
+    """
+    """
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    subscription_id = models.PositiveIntegerField(db_index=True)
```

Comment:
Are we expecting joins and a lot of data wrangling on this? If it's only to keep history then we can avoid indexing here.

Reply:
No we are not. removing